docs_src/microservices/device/details/DeviceProfiles.md

```diff
@@ -4,4 +4,126 @@ title: Device Service - Device Profiles
+which defines that device type in terms of the operations which it supports.
+
+Typically, device profiles are loaded from file by the Device Service and pushed to Core Metadata for storage on first start. 
+Once stored in Core Metadata, the device profiles are loaded from Core Metadata on subsequent starts.
+
+!!! warning
+    Editing the local device profile after Device Service has started will not impact the actual device profile being used. The old version must first be removed from Core Metadata and then restart the Device Service to push the new version to Core Metadata.
+
+Device profiles can also be manually Added/Updated/Deleted using Core Metadata's Device Profile [REST API](../../core/metadata/ApiReference.md)
+
+## Device Profile
+
+A device profile consists of the following fields.
+
+| Field Name      | Type                    | Required? | Description                                               |
+|:----------------|:------------------------|:----------|:----------------------------------------------------------|
+| name            | String                  | Y         | Must be unique in the EdgeX deployment.                   |
+| description     | String                  | N         | Description of the Device Profile                         |
+| manufacturer    | String                  | N         | Manufacture of the device described by the Device Profile |
+| model           | String                  | N         | Model of the device(s) described by the Device Profile    |
+| labels          | Array of String         | N         | Free form labels for querying device profiles             |
+| deviceResources | Array of DeviceResource | Y         | See [Device Resources](#device-resources) below           |
+| deviceCommands  | Array of DeviceCommand  | N         | See [Device Commands](#device-commands) below             |
+
+### Device Resources
+
+A device resource specifies a sensor value within a device that may be read
+from or written to either individually or as part of a device command. It has a
+name for identification and a description for informational purposes.
+
+The device resource consists of the following fields:
+
+| Field Name  | Type                 | Required? | Notes                                                                           |
+|:------------|:---------------------|:----------|:--------------------------------------------------------------------------------|
+| name        | String               | Y         | Must be unique in the EdgeX deployment.                                         |
+| description | String               | N         | Description of the device resource                                              |
+| isHidden    | Bool                 | N         | Do not expose the device resource via the Command Service or not, default false |
+| tag         | String               | N         |                                                                                 |
```

Comment:
Just wonder should `tag` be updated to `tags` and a `String-Interface Map` for type (see deviceResource [DTO](https://github.com/edgexfoundry/go-mod-core-contracts/blob/dc3ea5377182d585dff78450dfd9ececf0c61efa/dtos/deviceresource.go#L16))?